js/client-ladder.js

```diff
@@ -135,22 +139,31 @@
 				var self = this;
 				this.$el.html('<div class="ladder pad"><p><button name="selectFormat"><i class="fa fa-chevron-left"></i> Format List</button></p><p><em>Loading...</em></p></div>');
 				if (app.localLadder) {
-					app.send('/cmd laddertop ' + format);
+					app.send('/cmd laddertop ' + format + (this.curSearchVal ? ' ,' + this.curSearchVal : ''));
 				} else {
 					$.get('/ladder.php', {
 						format: format,
 						server: Config.server.id.split(':')[0],
-						output: 'html'
+						output: 'html',
+						prefix: this.curSearchVal
 					}, function (data) {
 						if (self.curFormat !== format) return;
-						var buf = '<div class="ladder pad"><p><button name="selectFormat"><i class="fa fa-chevron-left"></i> Format List</button></p><p><button class="button" name="refresh"><i class="fa fa-refresh"></i> Refresh</button></p>';
-						buf += '<h3>' + BattleLog.escapeFormat(format) + ' Top 500</h3>';
+						var buf = '<div class="ladder pad"><p><button name="selectFormat"><i class="fa fa-chevron-left"></i> Format List</button></p><p><button class="button" name="refresh"><i class="fa fa-refresh"></i> Refresh</button>';
+						buf += '<input type="text" id="ladderSearchBar" name="search" class="textbox searchinput" value="' + (this.curSearchVal || '') + '" placeholder="username prefix"/></p>';
+						buf += '<h3>' + BattleLog.escapeFormat(format) + ' Top ' + (this.curSearchVal ? "- '" + this.curSearchValue + "'" : '500') + '</h3>';
 						buf += data + '</div>';
 						self.$el.html(buf);
 					}, 'html');
 				}
 			}
 		},
+		searchChange: function (e) {
+			// 91 for right CMD / 93 for left CMD / 17 for CTL
```

Comment:
Why this set of detections? Why not just test `this.curSearchVal !== e.currentTarget.value`?

Reply:
I coped both of this (and below) from the teambuilder search. Not entirely sure why its the case either, which probably means I shouldnt have added it. But is it relevant in teambuilder even?

Reply:
The teambuilder search isn't implemented by me and so is worse than things implemented by me. ;)